baconTests/StorageManagerTests.swift

```diff
@@ -41,43 +36,153 @@ class StorageManagerTests: XCTestCase {
+        try! database.clearTransactionDatabase()
+        XCTAssertEqual(database.getNumberOfTransactionsInDatabase(), 0)
+        // Test loading empty database
+        XCTAssertTrue(try database.loadTransactions(limit: 10).isEmpty)
+        XCTAssertTrue(try database.loadTransactions(limit: 0).isEmpty)
+
+        // Save transactions of different type
+        let transactions = [TestUtils.validTransactionExpenditure04,
+                            TestUtils.validTransactionExpenditure03,
+                            TestUtils.validTransactionIncome02,
+                            TestUtils.validTransactionExpenditure01]
+        XCTAssertNoThrow(try database.saveTransaction(transactions[1]))
+        XCTAssertNoThrow(try database.saveTransaction(transactions[2]))
+        XCTAssertNoThrow(try database.saveTransaction(transactions[0]))
+        XCTAssertNoThrow(try database.saveTransaction(transactions[3]))
+        let loadedTransactions = try! database.loadTransactions(limit: 4)
+        XCTAssertEqual(loadedTransactions.count, 4)
+        // Check that the transactions loaded out are equal and in reverse chronological order
+        XCTAssertEqual(loadedTransactions, transactions)
+    }
 
-        // swiftlint:enable force_try
+    func test_invalid_loadTransactions_limit() {
+        let database = try! StorageManager()
+        XCTAssertThrowsError(try database.loadTransactions(limit: -1))
     }
 
-    func test_loadTransactions_OfType() {
-        // swiftlint:disable force_try
+    func test_loadTransactions_after() {
+        let database = try! StorageManager()
+        // Clear database
+        try! database.clearTransactionDatabase()
+        XCTAssertEqual(database.getNumberOfTransactionsInDatabase(), 0)
+        // Test loading empty database
+        XCTAssertTrue(try database.loadTransactions(after: TestUtils.january1st2019time0800, limit: 10).isEmpty)
+        XCTAssertTrue(try database.loadTransactions(after: TestUtils.january1st2019time0800, limit: 0).isEmpty)
+
+        // Save transactions
+        let transactions = [TestUtils.validTransactionDate03,
+                            TestUtils.validTransactionDate01point2]
+        XCTAssertNoThrow(try database.saveTransaction(TestUtils.validTransactionDate01))
+        XCTAssertNoThrow(try database.saveTransaction(TestUtils.validTransactionDate01point2))
+        XCTAssertNoThrow(try database.saveTransaction(TestUtils.validTransactionDate03))
+        let loadedTransactions = try! database.loadTransactions(after: TestUtils.january1st2019time0800, limit: 5)
+        XCTAssertEqual(loadedTransactions.count, 2)
+        // Check that the transactions loaded out are equal and in reverse chronological order
+        XCTAssertEqual(loadedTransactions, transactions)
+
+        // Test limit
+        let limitedTransactions = try! database.loadTransactions(after: TestUtils.january1st2019time0800, limit: 1)
+        XCTAssertEqual(limitedTransactions.count, 1)
+    }
```

Comment:
actually when you limit right, is the one immediately after the date that should be returned, or the most recent one (and also after the date) that should be returned? (should be the latter right?)
may test the actual content of the returned array also